internal/pkg/cli/task_run.go

```diff
@@ -310,12 +312,22 @@ func (o *runTaskOpts) Validate() error {
 		return errors.New("cannot specify both `--image` and `--dockerfile`")
 	}
 
+	if o.image != "" && o.isDockerfileContextSet {
+		return errors.New("cannot specify both `--image` and `--build-context`")
+	}
+
 	if o.isDockerfileSet {
 		if _, err := o.fs.Stat(o.dockerfilePath); err != nil {
 			return err
 		}
 	}
 
+	if o.isDockerfileContextSet {
+		if _, err := o.fs.Stat(o.dockerfileContextPath); err != nil {
+			return err
+		}
+	}
```

Comment:
After https://github.com/aws/copilot-cli/pull/3046/files#r751679516 we could simply have
```suggestion
	if _, err := o.fs.Stat(filepath.Join(o.dockerfileContextPath, o.dockerfilePath)); err != nil {
		return err
	}
```
to check if the dockerfile exists.

Reply:
It's possible that the Dockerfile and Context and in completely separate directories, that's why the stat checks are separate